spec/controllers/api/v1/restaurants_controller_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe Api::V1::RestaurantsController, type: :controller do
+  let!(:first_user) {User.create!(id: 1, first_name: "John", last_name: "Smith", username: "johnsmith", email: "johnsmith@smith.com", password: "1234567" )}
+  let!(:first_restaurant) { Restaurant.create!( name: "Panera", address: "55 Summer St.", city: "Boston",
+          state: "MA", zip: "02111", dollar_value: 2, description: "something", picture: "fakephoto", phone_number: "1234567890",
+          website_url: "asdf", user_id: 1) }
+
+  describe "GET#index" do
+    it "should return a single sample from all restaurants" do
+      get :index
+      returned_json = JSON.parse(response.body)
+      expect(response.status).to eq 200
+      expect(response.content_type).to eq("application/json")
+
+      expect(returned_json.length).to eq 1
+      expect(returned_json["restaurant"]["name"]).to eq "Panera"
+      expect(returned_json["restaurant"]["city"]).to eq "Boston"
+      expect(returned_json["restaurant"]["dollar_value"]).to eq 2
+    end
+  end
+
+  describe "POST#create" do
+    it "should create a new restaurant" do
+
+      post_json = { name: "Panera", address: "55 Summer St.", city: "Boston",
+              state: "MA", zip: "02111", dollar_value:
+              2, description: "something", picture: "fakephoto", phone_number: "1234567890",
+              website_url: "asdf", user_id: 1 }.to_json
+
+      expect{ post :create, body: post_json }.to change{ Restaurant.count }.by(1)
+    end
+
+    it "should return the json of the new restaurant" do
+
+      post_json = { name: "Panera", address: "55 Summer St.", city: "Boston",
+              state: "MA", zip: "02111", dollar_value:
+              2, description: "something", picture: "fakephot", phone_number: "1234567890",
+              website_url: "asdf", user_id: 1 }.to_json
+
+      post :create, body: post_json
+      returned_json = JSON.parse(response.body)["restaurant"]
+      expect(response.status).to eq 200
+      expect(response.content_type).to eq("application/json")
+
+      expect(returned_json).to be_kind_of(Hash)
+      expect(returned_json).to_not be_kind_of(Array)
+      expect(returned_json["name"]).to eq "Panera"
+
+
+    end
+  end
+end
```

Comment:
I love API tests! This one could be cleaned up a little (especially if you used `FactoryGirl`) but I am happy to see it either way.